gbpservice/nfp/configurator/drivers/firewall/asav/asav_fw_driver.py

```diff
@@ -0,0 +1,960 @@
+        be input through CLI.
+
+        :param mgmt_ip: Management IP address of the Service VM
+        :param commands: list of commands
+        :param response_data_expected: If set, the REST call to the Service VM
+        returns the result of the call to the caller.
+
+        Returns: Result of the REST call.
+
+        """
+
+        resource_uri = "/api/cli"
+        url = const.ASAV_REQUEST_URL % (mgmt_ip, resource_uri)
+
+        if not response_data_expected or not commands:
+            commands.append('write memory')
+        data = {"commands": commands}
+
+        return self.rest_api.post(url, data, self.auth, response_data_expected)
+
+    def _get_interface_commands(self, interface_name, interface_index,
+                                interface_ip, interface_net_mask,
+                                security_level,
+                                asav_interface_type='gigabitEthernet',
+                                mac_address=None):
+        """ Calculates the interface position.
+
+        :param interface_name: Name of the interface
+        :param interface_index: Position of the interface
+        :param interface_ip: IP address of the interface
+        :param interface_net_mask: Subnet mask of the interface
+        :param security_level: Security level to be associated with
+        the interface
+        :param asav_interface_type: Management/Gigabitethernet
+        :param mac_address: MAC address of the interface
+
+        Returns: A list of commands to configure interface.
+
+        """
+
+        commands = list()
+        commands.append("interface " + asav_interface_type + " 0/" +
+                        interface_index)
+        commands.append("nameif " + interface_name)
+        commands.append("security-level " + security_level)
+        command = "ip address " + interface_ip + " " + interface_net_mask
+
+        commands.append(command)
+        commands.append("no shutdown")
+
+        allow_inter_interface_traffic = ("same-security-traffic permit "
+                                         "inter-interface")
+        commands.append(allow_inter_interface_traffic)
+        return commands
+
+    def get_interface_position(self, mgmt_ip, mac):
+        """ Calculates the interface position.
+
+        :param mgmt_ip: Management IP address of the Service VM
+        :param mac: MAC address of the VM for which the
+        interface position has to be found.
+
+        Returns: interface index.
+
+        """
+
+        commands = list()
+
+        commands.append("sh inte")
+        result = self.configure_bulk_cli(mgmt_ip, commands,
+                                         response_data_expected=True)
+
+        if result.get('GET_RESPONSE'):
+            data = ''.join(result['GET_RESPONSE']['response']).split(
+                                                        'GigabitEthernet0/')
+            for item in data:
+                if mac in item:
+                    return item[0]
```

Comment:
Will ASAV list the interfaces in order? If not, this logic will break. Ideal approach will be to get the index from "GigabitEthernet0/<Index>" rather than the item position right?